pocs/focuser/focuslynx.py

```diff
@@ -0,0 +1,327 @@
+
+    def connect(self):
+        try:
+            # Configure serial port.
+            self._serial_port = serial.Serial(port=self.port,
+                                              baudrate=115200,
+                                              bytesize=serial.EIGHTBITS,
+                                              parity=serial.PARITY_NONE,
+                                              stopbits=serial.STOPBITS_ONE,
+                                              timeout=1.0)
+
+        except serial.SerialException as err:
+            self._serial_port = None
+            self.logger.critical('Could not open {}!'.format(self.port))
+            raise err
+
+        self.logger.debug('Established serial connection to {} on {}'.format(self.name, self.port))
+
+    def move_to(self, position, blocking=True):
+        """
+        Move encoder to a new encoder position. Blocking by default, but if blocking=False is
+        passed then will return immediately. If blocking=True returns actual position following
+        move, otherwise returns target position
+        """
+        if position < self._min_position:
+            message = 'Requested position {} less than min position, moving to {}!'.format(position,
+                                                                                           self._min_position)
+            self.logger.error(message)
+            warn(message)
+            position = self._min_position
+        elif position > self._max_position:
+            message = 'Requested position {} greater than max position, moving to {}!'.format(position,
+                                                                                              self._max_position)
+            self.logger.error(message)
+            warn(message)
+            position = self._max_position
+
+        self.logger.debug('Moving focuser {} to {}'.format(self.uid, position))
+        self._send_command('<F{:1d}MA{:06d}>'.format(self._focuser_number, position),
+                           command_type='MOVE')
+
+        if blocking:
+            while self.is_moving:
+                time.sleep(1)
+            if self.position != self._target_position:
+                message = "Focuser {} did not reach target position {}, now at {}!".format(self.uid,
+                                                                                           self._target_position,
+                                                                                           self._position)
+                self.logger.warning(message)
+                warn(message)
+            return self._position
+        else:
+            return position
+
+    def halt(self):
+        """
+        Causes the focuser to immediately stop any movements
+        """
+        self._send_command('<F{:1d}HALT>'.format(self._focuser_number), command_type='HALT')
+        self.logger.warning("Focuser {} halted".format(self.uid))
```

Comment:
Maybe need to run `_update_focuser_status()` after this?

Reply:
Probably should, yes.  Kind of inclined to `warning.warn()` here too, as I basically see `halt()` being the emergency stop button, which should have plenty of user feedback.